R/checks.R

```diff
@@ -1,101 +1,107 @@
+  msg <- c(
+    "Desirability values should be numeric and complete in the range [0, 1].",
+    "i" = "Current values are {.obj_type_friendly {x}}.",
+    "i" = "{?Some v/V}alue{?s} {?are/is} outside the [0, 1] range."
+  )
+
+  if (!is.vector(x) || !is.numeric(x)) {
+    cli::cli_abort(msg, call = call)
   }
 
-  if (length(high) != 1 || !is.numeric(high) || is.na(high)) {
-    rlang::abort("'high' should be a single numeric value.")
+  if (out_of_unit_range(x) | length(x) > 1) {
+    cli::cli_abort(msg, call = call)
   }
 
+  invisible(NULL)
+}
+
+check_value_order <- function(low, high, target = NULL, call) {
+  check_number_decimal(low, call = call)
+  check_number_decimal(high, call = call)
+  check_number_decimal(target, allow_null = TRUE, call = call)
+
   if (!is.null(target)) {
-    if (length(target) != 1 || !is.numeric(target) || is.na(target)) {
-      rlang::abort("'target' should be a single numeric value.")
-    }
     ord <- low < target & target < high
     if (!ord) {
-      rlang::abort("The values should be `low < target < high`.")
-    }
-  } else {
-    ord <- low < high
-    if (!ord) {
-      rlang::abort("The values should be `low < high`.")
+      cli::cli_abort("The values should be {.code low < target < high}  (actual
+                     are {low}, {target}, and {high}).", call = call)
     }
   }
+
+  ord <- low < high
+  if (!ord) {
+    cli::cli_abort("The values should be {.code low < high} (actual are {low}
+                   and {high}).", call = call)
+  }
+
   invisible(NULL)
 }
 
-is_vector_args <- function(values, d) {
+is_vector_args <- function(values, d, call) {
   if (!is.vector(values) || !is.numeric(values)) {
-    rlang::abort("'values' should be a numeric vector.")
+    cli::cli_abort("{.arg values} should be a numeric vector.", call = call)
   }
   if (!is.vector(d) || !is.numeric(d)) {
-    rlang::abort("'d' should be a numeric vector.")
+    cli::cli_abort("'d' should be a numeric vector.", call = call)
   }
   if (length(values) != length(d)) {
-    rlang::abort("'values' and 'd' should be the same length.")
+    cli::cli_abort("'{.arg values}' and '{.arg d}' should be the same length.",
+                   call = call)
   }
   invisible(TRUE)
 }
 
 
-check_args <- function(arg, x, use_data, fn, type = "low") {
+check_args <- function(arg, x, use_data, fn, type = "low", call) {
   if (rlang::is_missing(arg)) {
     if (use_data) {
-      type <- rlang::arg_match0(type, c("low", "high", "target"))
+      type <- rlang::arg_match0(type, c("low", "high", "target"), error_call = call)
       .fn <- switch(type, low = min, high = max, target = stats::median)
       arg <- .fn(x, na.rm = TRUE)
     } else {
-      rlang::abort(
-        glue::glue("In `{fn}()`, argument '{type}' is required when 'new_data = FALSE'.")
-      )
+      cli::cli_abort("In {.fn {fn}}, argument {.arg {type}} is required when
+                     {.code new_data = FALSE}.", call = call)
```

Comment:
More of a general note. Since we are passing the `call`, we don't need the `fn` argument anymore.